From: Geert Uytterhoeven <geert@linux-m68k.org>
Date: Mon, 11 Aug 2014 21:42:49 +0200
Subject: m68k: Wire up memfd_create
Origin: https://git.kernel.org/linus/4ed7800987b1b082f8fc98c5cb7eb20cf74280a8
Bug-Debian: https://bugs.debian.org/760702

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
[bwh: Backported to 3.16:
 - Adjust context
 - Insert unimplemented-syscall entry for getrandom]
---
--- a/arch/m68k/include/asm/unistd.h
+++ b/arch/m68k/include/asm/unistd.h
@@ -4,7 +4,7 @@
 #include <uapi/asm/unistd.h>
 
 
-#define NR_syscalls		352
+#define NR_syscalls		354
 
 #define __ARCH_WANT_OLD_READDIR
 #define __ARCH_WANT_OLD_STAT
--- a/arch/m68k/include/uapi/asm/unistd.h
+++ b/arch/m68k/include/uapi/asm/unistd.h
@@ -357,5 +357,6 @@
 #define __NR_sched_setattr	349
 #define __NR_sched_getattr	350
 #define __NR_renameat2		351
+#define __NR_memfd_create	353
 
 #endif /* _UAPI_ASM_M68K_UNISTD_H_ */
--- a/arch/m68k/kernel/syscalltable.S
+++ b/arch/m68k/kernel/syscalltable.S
@@ -372,4 +372,6 @@ ENTRY(sys_call_table)
 	.long sys_sched_setattr
 	.long sys_sched_getattr		/* 350 */
 	.long sys_renameat2
+	.long sys_ni_syscall		/* getrandom */
+	.long sys_memfd_create
 
